### PR TITLE
Added comments and suggested refactorings

### PR DESCRIPTION
**Pull request by jelledruyts**

### CODE REVIEW

File: TodoListService/App_Start/Startup.Auth.cs [hunk 1]
```diff
@@ -30,6 +30,7 @@ public void ConfigureAuth(IAppBuilder app)
             app.UseActiveDirectoryFederationServicesBearerAuthentication(
                 new ActiveDirectoryFederationServicesBearerAuthenticationOptions
                 {
+                    // [jelled] Why is Audience not being checked via this mechanism?
```
Comment (tvuylsteke) [inline TodoListService/App_Start/Startup.Auth.cs:33]: [tvuylsteke] 4 lines above the validAudience is populated with the configuration file version. I should refactor this. Pointless usage of variables.

File: TodoListService/Controllers/ClaimsController.cs [hunk 2]
```diff
@@ -25,7 +25,7 @@ public IEnumerable<ClaimInfo> Get()
                 ci.ClaimType = claim.Type;
                 ci.Value = claim.Value;
                 ci.ValueType = claim.ValueType.Substring(claim.ValueType.IndexOf('#') + 1);
-                ci.SubjectName = ((claim.Subject) != null && (String.IsNullOrEmpty(claim.Subject.Name)) == false) ? claim.Subject.Name : "Null";
+                ci.SubjectName = (claim.Subject == null || string.IsNullOrEmpty(claim.Subject.Name)) ? "Null" : claim.Subject.Name;
```
Comment (tvuylsteke) [inline TodoListService/Controllers/ClaimsController.cs:28]: Good optimization. Your version is way less complex and eassier to read.

File: TodoListService/Models/ClaimInfo.cs [hunk 3]
```diff
@@ -7,8 +7,6 @@ namespace TodoListService.Models
 {
     public class ClaimInfo
     {
-        public ClaimInfo() { }
```
Comment (tvuylsteke) [inline TodoListService/Models/ClaimInfo.cs:10]: Is it common not to have an empty constructor? is this one available implicetely? Why remove this one?

File: TodoListWebApp/Controllers/HomeController.cs [hunk 4]
```diff
@@ -36,81 +36,84 @@ public ActionResult Index()
...
+                        // [jelled] Show the error somehow.
                     }
-
-                    ViewBag.ATclaims = ATresults;
                 }
-                else
+                catch (AdalSilentTokenAcquisitionException)
                 {
-                    ViewBag.ATclaims = new List<ClaimInfo>();
+                    // [jelled] In case the exception is because acquiring the token silently failed, then the cached token is probably invalid
```
Comment (tvuylsteke) [inline TodoListWebApp/Controllers/HomeController.cs:108]: I fully agree. And for the actual TodoList page this is the case. But from a demonstration perspective I want this page to only show the claims the user has. If he doesn't have any, I'm not showing those. Theoretically spoken, I could have an ID Token here, but not an Access Token. Then I want this page to reflect that.

File: TodoListService/Models/ClaimInfo.cs (hunk 3 above)
Comment (jelledruyts) [inline TodoListService/Models/ClaimInfo.cs:10]: Indeed, if there are no constructors defined, the compiler always emits a default (empty) one. So it's common to not specify any for simple "data" classes such as this one.

**Pull request closed without merging**